### ISSUE TITLE
about loss function

### ISSUE THREAD

**Issue by pch9520** [t2kasa/social_lstm_keras_tf#8]
```
def _compute_loss(self, y_batch, o_batch):
        """
        :param y_batch: (batch_size, pred_len, max_n_peds, pxy_dim)
        :param o_batch: (batch_size, pred_len, max_n_peds, out_dim)
        :return: loss
        """
        not_exist_pid = 0

        y = tf.reshape(y_batch, (-1, pxy_dim))
        o = tf.reshape(o_batch, (-1, out_dim))

        pids = y[:, 0]

        # remain only existing pedestrians data
        exist_rows = tf.not_equal(pids, not_exist_pid)
        y_exist = tf.boolean_mask(y, exist_rows)
        o_exist = tf.boolean_mask(o, exist_rows)
        pos_exist = y_exist[:, 1:]

        # compute 2D normal prob under output parameters
        log_prob_exist = normal2d_log_pdf(o_exist, pos_exist)
        # for numerical stability
        log_prob_exist = tf.minimum(log_prob_exist, 0.0)

        loss = -log_prob_exist
        return loss
```
I look into this function, finally the loss type is (6*52, 1), loss can be a vector but not a scalar?

**Comment by pch9520** [comment]
I think I have solved my problem, although the loss is a vector, but if the loss weights are all one, the loss just equal K.sum the vector?
